include/private/entry.h

```diff
@@ -27,24 +27,137 @@
 #  include <stumpless/severity.h>
 #  include "private/strbuilder.h"
 
+/**
+ * Frees entry cache
+ *
+ * **Thread Safety: MT-Unsafe**
+ * This function is not thread safe as it destroys resources(entry cache)
+ * that other threads can use.
+ *
+ * **Async Signal Safety: AS-Unsafe**
+ * This function is not safe to call from signal handlers due to destroying
+ * resources(entry cache) that might be used within a function.
+ *
+ * **Async Cancel Safety: AC-Unsafe**
+ * This function is not safe to call from threads that may be asynchronously
+ * cancelled, due to leaving memory in an undefined state.
+ *
+ * @since release v1.6.0
+ *
+ */
 void
 entry_free_all( void );
 
+/**
+ * Gets the priority value from facility and severity parameters.
+ * 
+ * Priority value is calculated by left shifting the facility value by 3 
+ * and adding it to the severity value which is according to the 
+ * RFC 5424 Section 6.2.1. The shift operation is done prior to get_prival()
+ * function with macros in "facility.h"
+ *
+ * **Thread Safety: MT-Safe**
+ * This function is thread safe.
+ *
+ * **Async Signal Safety: AS-Safe **
+ * This function must be safe to call from signal handlers
+ *
+ * **Async Cancel Safety: AC-Safe**
+ * This function must be safe to call from threads that may be asynchronously
+ * cancelled.
+ *
+ * @since release v2.0.0
+ *
+ * @param facility Facility value. This should be a \c STUMPLESS_FACILITY value.
+ * 
+ * @param severity Severity value. This should be a \c STUMPLESS_SEVERITY value
+ *
+ * @return Priority value
+ */
 int
 get_prival( enum stumpless_facility facility,
             enum stumpless_severity severity );
 
+/**
+ * Locks the mutex within the entry.
+ *
+ * **Thread Safety: MT-Safe**
+ * This function is thread safe.
+ *
+ * **Async Signal Safety: AS-Unsafe **
+ * This function is not safe to call from signal handlers due to the use of
+ * function 'pthread_mutex_lock()' from <pthread.h> that is not guarenteed to 
```

Comment:
Some builds, for example Windows, don't use the `pthread.h` functions, but instead use the Windows mutex libraries. Simplify this sentence to simply mention "mutexes" and leave out the implementation detail.